drivers/spi/spi_context.h

```diff
@@ -112,13 +112,29 @@ static inline void spi_context_release(struct spi_context *ctx, int status)
 static inline int spi_context_wait_for_completion(struct spi_context *ctx)
 {
 	int status = 0;
+	uint64_t ms;
+
+	ms = MAX(ctx->tx_len, ctx->rx_len) * 8 * 1000 / ctx->config->frequency;
+	ms += 200; /* tolerance */
```

Comment:
Why 200ms, why hard-coded and why is it not optional that this change does not break anything?

Reply:
200ms is somewhat the high tolerance level which could be sufficient for most of the SPI controllers available in market. This is what the Linux kernel is using today and I took inspiration from it:
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/spi/spi.c?id=833bfade96561216aa2129516a5926a0326860a2

For sure this change won't break anything. If it does then most likely we need to increase the tolerance level.

Reply:
> For sure this change won't break anything.

really?, you change behavior of each SPI driver,  and use hard-coded 200ms value.

Reply:
Yes, this impacts every SPI driver but it won't cause any issue IMO as we have this same logic working in Linux kernel for years.

Reply:
@jfischer-no Moreover the current implementation of waiting forever won't work in all cases where the slave is powered down (see linked issue). So the timeout logic makes use of the transfer speed plus 200ms tolerance value.